common/src/main/java/com/microsoft/identity/common/internal/cache/MsalRefreshTokenCacheItem.java

```diff
@@ -0,0 +1,98 @@
+import com.microsoft.identity.common.Account;
+import com.microsoft.identity.common.internal.providers.azureactivedirectory.AzureActiveDirectory;
+import com.microsoft.identity.common.internal.providers.azureactivedirectory.AzureActiveDirectoryAccount;
+import com.microsoft.identity.common.internal.providers.azureactivedirectory.AzureActiveDirectoryAuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.azureactivedirectory.AzureActiveDirectoryCloud;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenResponse;
+import com.microsoft.identity.common.internal.util.EncodingUtil;
+
+import java.net.URL;
+
+/**
+ * A lightweight representation of MSAL's cache item for refresh tokens.
+ * <p>
+ * This object knows how to create a cache key for itself.
+ * <p>
+ * If you call getCacheValue() on this object you get it as JSON.
+ */
+class MsalRefreshTokenCacheItem extends BaseMsalTokenCacheItem implements ISelfSerializingCacheItem {
+
+    @SerializedName("refresh_token")
+    private String mRefreshToken;
+
+    @SerializedName("environment")
+    private String mEnvironment;
+
+    // meta data used to construct user object from refresh token cache item.
+    @SerializedName("displayable_id")
+    String mDisplayableId;
+
+    @SerializedName("name")
+    String mName;
+
+    @SerializedName("identity_provider")
+    String mIdentityProvider;
+
+    transient String mUserIdentifier;
+
+    /**
+     * Constructs a new MsalRefreshTokenCacheItem.
+     *
+     * @param oAuth2Strategy The Strategy of this IdP.
+     * @param request        The {@link AuthorizationRequest} placed.
+     * @param response       The {@link TokenResponse} of this request.
+     */
+    public MsalRefreshTokenCacheItem(
+            OAuth2Strategy oAuth2Strategy,
+            AuthorizationRequest request,
+            TokenResponse response) {
+        super(oAuth2Strategy, request, response);
+        final Account account = oAuth2Strategy.createAccount(response);
+        mRefreshToken = response.getRefreshToken();
+
+        if (request instanceof AzureActiveDirectoryAuthorizationRequest) {
+            final URL authority = ((AzureActiveDirectoryAuthorizationRequest) request).getAuthority();
+            final AzureActiveDirectoryCloud cloudEnv = AzureActiveDirectory.getAzureActiveDirectoryCloud(authority);
+            // This map can only be consulted if authority validation is on.
+            // If the host has a hardcoded trust, we can just use the hostname.
+            if (null != cloudEnv) {
+                mEnvironment = cloudEnv.getPreferredNetworkHostName();
+            } else {
+                mEnvironment = authority.getHost();
+            }
+        }
+
+        if (account instanceof AzureActiveDirectoryAccount) {
```

Comment:
In the case of MSAL I don't believe that you need all of these identifiers... there's an abstract method on the account object for getting the unique identifier